### PR TITLE
Reworked Add Liquidity modal

### LINKED ISSUE #283: The second token balance in add liquidity is not checked
Reported by jrafaelangarita#2400 https://discord.com/channels/844177080005951489/963084712430174269/1070758692594655262

### LINKED ISSUE #286: ADD LIQUIDITY from wallet page not functioning correctly
2 issues I have identified here:

1. TNKR asset in wallet doesn't have `ADD LIQUIDITY` option, despite there being an active TNKR<>BSX pool
image

2. Probably most easily explained with the video - I am not sure what the logic is here when I select `ADD LIQUIDITY`, on the following modal I'm not really sure how an option becomes selectable or not - I should be able to add liquidity for any 2 assets.

https://user-images.githubusercontent.com/97540577/216560386-cf9f5173-248b-4887-9893-70713644e600.mov

### LINKED ISSUE #288: Add liquidity asset selector behaviour
Default logic:
As user, i want to be able to provide liquidity for selected pool
If i click on "Add liquidity" button on pools
I want to see modal window for add liq screen
And asset selectors should be prefilled with token combination based on from which pool user selected.
For example if user selected add liquidity on BSX/KSM pool, asset selectiors should be prefilled with BSX and KSM assets.

**Logic when user change one of the assets:**
If user change one of assets in asset selector, than second asset input should be reset and in dropwodn menu there should be displayed just assets which has pool with this token.
When user choose one of these tokens, it should be filled into asset selector.
Order should not change, ie BSX/aUSD and aUSD/BSX is the same

### LINKED ISSUE #297: Do not prefill second asset when user wants to provide liquidity going from wallet
Current behaviour:
Given user is in asset list
And Add liquidity functionality is available
When user click on "Add liquidity" button
Than Add liquidity modal is being displayed
And both assets are prefilled (based on some logic...)

Expected behaviour:
Just first asset(asset for which user selected add liquidity) should be prefilled.
Second asset dropdown should be empty, and when user click on asset dropdown, just assets with which first asset has pools should be listed.

Screen Recording 2023-02-06 at 15.44.03.mov

**Designs:**

### PR DESCRIPTION
**Pull request by vkulinich-cl**
Fixes #283
Fixes #286
Fixes #288 
Fixes #297

### CODE REVIEW

**Comment by jak-pan** [comment]
We're still missing the validations for balance (283)
Screenshot 2023-02-08 at 10 58 37

I would request one small change -> once we have validation on the add liquidity modal (disable add liq when not enough balance) I would like to enable add liquidity button on the pools page so the user can check how much of assets he needs to provide liquidity in specific pool (RN it's disabled if you don't have one of the 2 assets)

File: src/components/AssetSelect/AssetSelectInput.tsx [hunk 1]
```diff
@@ -65,6 +66,7 @@ export const AssetSelectInput = (props: {
           </Trans>
         </Text>
         <TokenInputMaxButton
+          disabled={!!props.disabled}
```
Comment (tomashapl1) [inline src/components/AssetSelect/AssetSelectInput.tsx:69]: `props.disabled`

**Comment by jak-pan** [comment]
cc @DuriSK

File: src/sections/assets/AssetsModal.tsx [hunk 2]
```diff
@@ -28,15 +34,21 @@ export const AssetsModal: FC<AssetsModalProps> = ({
...
+        }
 
-  const otherAssets =
-    (allowedAssets != null
-      ? assetsRows.data?.filter((asset) => !allowedAssets?.includes(asset.id))
-      : []) ?? []
+        return acc
+      },
+      { isPair: [] as AccountDetailsType, notPair: [] as AccountDetailsType },
+    ) ?? {}
```
Comment (tomashapl1) [inline src/sections/assets/AssetsModal.tsx:51]:
```
?? {
      isPair: [],
      notPair: []
    }
```

File: src/sections/assets/AssetsModalRow.styled.ts [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import styled from "@emotion/styled"
 import { theme } from "../../theme"
+import { css } from "@emotion/react"
 
-export const SAssetRow = styled.div`
+export const SAssetRow = styled.div<{ noPair?: boolean }>`
```
Comment (tomashapl1) [inline src/sections/assets/AssetsModalRow.styled.ts:5]: `notPair` to be consistent

File: src/sections/assets/AssetsModalRow.tsx [hunk 4]
```diff
@@ -14,10 +14,15 @@ import { DollarAssetValue } from "components/DollarAssetValue/DollarAssetValue"
...
 interface AssetsModalRowProps {
   id: Maybe<u32 | string>
+  noPair?: boolean
   onClick?: () => void
 }
 
-export const AssetsModalRow: FC<AssetsModalRowProps> = ({ id, onClick }) => {
+export const AssetsModalRow: FC<AssetsModalRowProps> = ({
+  id,
+  noPair,
```
Comment (tomashapl1) [inline src/sections/assets/AssetsModalRow.tsx:23]: same here

File: src/sections/pools/pool/modals/addLiquidity/PoolAddLiquidityModal.tsx [hunk 5]
```diff
@@ -34,20 +36,44 @@ import {
...
 interface PoolAddLiquidityModalProps {
-  pool: PoolBase
+  tradeFee?: PoolFee
   setPoolAddress: (address: string) => void
+  assetA: PoolToken | AssetMetaType
+  assetB: PoolToken | undefined
+  poolAddress: string
 }
 
+const getAllowedTokensId = (
```
Comment (tomashapl1) [inline src/sections/pools/pool/modals/addLiquidity/PoolAddLiquidityModal.tsx:48]: could we add it to `.utils.ts` ?

File: src/sections/pools/pool/modals/addLiquidity/PoolAddLiquidityModal.tsx [hunk 6]
```diff
@@ -57,26 +83,44 @@ export const PoolAddLiquidityModal: FC<PoolAddLiquidityModalProps> = ({
...
   const reserves = useTokensBalances(
-    pool.tokens.map((i) => i.id),
-    pool.address,
+    [assets.assetA.id, assets.assetB.id],
+    poolAddress,
   )
-  const balances = useTokensBalances(
-    pool.tokens.map((i) => i.id),
+
+  const [{ data: assetABalance }, { data: assetBBalance }] = useTokensBalances(
```
Comment (tomashapl1) [inline src/sections/pools/pool/modals/addLiquidity/PoolAddLiquidityModal.tsx:122]:
```
const balances = useTokensBalances(
```

to be consistent as `reserves`

File: src/sections/pools/pool/modals/addLiquidity/PoolAddLiquidityModal.tsx [hunk 7]
```diff
@@ -253,48 +297,99 @@ export const PoolAddLiquidityModal: FC<PoolAddLiquidityModalProps> = ({
     })
   }
 
+  const getValidationRules = (balance: BigNumber, decimals: string) => ({
```
Comment (tomashapl1) [inline src/sections/pools/pool/modals/addLiquidity/PoolAddLiquidityModal.tsx:300]: to utils please 🙏

**Review by tomashapl1 — CHANGES_REQUESTED** [review]

**Comment by jak-pan** [comment]
I think we're rounding inputs here which shouldn't be case and missing the err msg still

Screenshot 2023-02-08 at 13 15 05

Screenshot 2023-02-08 at 13 15 14

**Comment by jak-pan** [comment]
Ok scratch the second thing. It's good I just had too big balance :)

**Comment by jak-pan** [comment]
> We're still missing the validations for balance (#283) Screenshot 2023-02-08 at 10 58 37
> 
> I would request one small change -> once we have validation on the add liquidity modal (disable add liq when not enough balance) I would like to enable add liquidity button on the pools page so the user can check how much of assets he needs to provide liquidity in specific pool (RN it's disabled if you don't have one of the 2 assets)

This is fixed

**Review by tomashapl1 — APPROVED** [review]

**Pull request merged**